### PR TITLE
Updated to only use Credentials Management when it releases passwords plain text

### PR DESCRIPTION
**Pull request by TMSCH**
It uses the `preventSilentAccess` method to detect this.

### CODE REVIEW

File: ts/api/navigator_credentials.ts [hunk 1]
```diff
@@ -263,7 +265,14 @@ export class NavigatorCredentials implements OpenYoloApi {
  * it will create a no-op version of the API.
  */
 export function createNavigatorCredentialsApi(): OpenYoloApi {
-  if (navigator.credentials !== undefined && isSecureOrigin()) {
+  // As per
+  // https://developers.google.com/web/updates/2017/06/credential-management-updates#feature_detection_needs_attention,
+  // we check for the presence of preventSilentAccess in navigator.credentials
+  // that is a fewture of M60+. Previous versions would not release plain-text
```
Comment (iainmcgin) [inline ts/api/navigator_credentials.ts]: nit: fewture -> feature

File: ts/api/navigator_credentials_test.ts [hunk 2]
```diff
@@ -51,14 +51,13 @@ describe('NavigatorCredentials', () => {
       const options: OpenYoloCredentialRequestOptions = {
         supportedAuthMethods: [AUTHENTICATION_METHODS.GOOGLE]
       };
-      const federatedCredential: FederatedCredential = {
+      const federatedCredential = new FederatedCredential({
         provider: AUTHENTICATION_METHODS.GOOGLE,
-        protocol: null,
+        protocol: 'protocol',
```
Comment (iainmcgin) [inline ts/api/navigator_credentials_test.ts]: Based on the spec, this should be 'openidconnect' or null.

File: ts/api/navigator_credentials.ts (hunk 1 above)
Comment (TMSCH) [inline ts/api/navigator_credentials.ts]: Done.

File: ts/api/navigator_credentials_test.ts (hunk 2 above)
Comment (TMSCH) [inline ts/api/navigator_credentials_test.ts]: Done.

**Comment by TMSCH** [comment]
Thanks Iain!

**Pull request merged**